packages/legacy/core/App/screens/ProofDetails.tsx

```diff
@@ -0,0 +1,242 @@
+      <View style={styles.container}>
+        <View style={styles.header}>
+          <View style={styles.headerTitleContainer}>
+            <CheckInCircle {...{ height: 45, width: 45 }} />
+            <Text style={styles.headerTitle}>{t('Verifier.InformationReceived')}</Text>
+          </View>
+          <Text style={styles.headerDetails}>
+            {connectionLabel} {t('Verifier.InformationReceivedDetails')}
+          </Text>
+        </View>
+        <View style={styles.content}>
+          <SharedProofData recordId={record.id} />
+        </View>
+        <View style={styles.footerButton}>
+          <Button
+            title={t('Verifier.GenerateNewQR')}
+            accessibilityLabel={t('Verifier.GenerateNewQR')}
+            testID={testIdWithKey('GenerateNewQR')}
+            buttonType={ButtonType.Primary}
+            onPress={onGenerateNew}
+          />
+        </View>
+      </View>
+    </ScrollView>
+  )
+}
+
+const UnverifiedProof: React.FC<UnverifiedProofProps> = ({ record }) => {
+  const { t } = useTranslation()
+  const { ColorPallet } = useTheme()
+
+  const styles = StyleSheet.create({
+    header: {
+      flexGrow: 1,
+      backgroundColor: ColorPallet.semantic.error,
+      paddingHorizontal: 30,
+      paddingVertical: 20,
+    },
+    headerTitleContainer: {
+      flexDirection: 'row',
+      justifyContent: 'flex-start',
+      alignItems: 'center',
+    },
+    headerTitle: {
+      marginHorizontal: 8,
+      color: ColorPallet.grayscale.white,
+      fontSize: 34,
+      fontWeight: 'bold',
+    },
+    headerDetails: {
+      color: ColorPallet.grayscale.white,
+      marginVertical: 10,
+      fontSize: 18,
+    },
+  })
+
+  return (
+    <View testID={testIdWithKey('UnverifiedProofView')}>
+      <View style={styles.header}>
+        <View style={styles.headerTitleContainer}>
+          <Icon name="bookmark-remove" size={45} color={'white'} />
+          {record.state === ProofState.Abandoned && (
+            <Text style={styles.headerTitle}>{t('Verifier.PresentationDeclined')}</Text>
+          )}
+          {record.isVerified === false && (
+            <Text style={styles.headerTitle}>{t('Verifier.ProofVerificationFailed')}</Text>
+          )}
+        </View>
+      </View>
+    </View>
+  )
+}
+
+const ProofDetails: React.FC<ProofDetailsProps> = ({ route, navigation }) => {
+  if (!route?.params) {
+    throw new Error('ProofRequesting route prams were not set properly')
+  }
+
+  const { recordId, isHistory } = route?.params
+  const record: ProofExchangeRecord = useProofById(recordId)
+  const { agent } = useAgent()
+
+  useEffect(() => {
+    if (agent) markProofAsViewed(agent, record)
+  }, [])
+
+  useFocusEffect(
+    useCallback(() => {
+      const onBackPress = () => {
+        if (route.params.isHistory) {
+          navigation.goBack()
+        } else {
+          navigation.navigate(Screens.ProofRequests, {})
+        }
+        return true
```

Comment:
Is this return needed?

Reply:
Linter warns that return type must be either boolean or undefined or null.  If we return this statement the return type will be void